packages/tracing/src/Tracer.ts

```diff
@@ -261,7 +261,12 @@ class Tracer implements TracerInterface {
       return this.provider.captureAWSClient(service);
     } catch (error) {
       try {
-        return this.provider.captureAWSClient((service as unknown as T & { service: T }).service);
+        // This is needed because some aws-sdk clients like AWS.DynamoDB.DocumentDB don't comply with the same
+        // instrumentation contract like most base clients. 
+        // For detailed explanation see: https://github.com/awslabs/aws-lambda-powertools-typescript/issues/524#issuecomment-1024493662
+        this.provider.captureAWSClient((service as unknown as T & { service: T }).service);
```

Comment:
Thanks for the detailed investigation!

Reply:
@dreamorosi  question just for myself.

Why do we cast to `unknown` first and as `T`  in the 2nd step?

Reply:
You are right, it's a leftover from a previous experiment and it's not needed since `T` is already a generic. Fixed in latest commit.